src/main/resources/com/google/api/codegen/clientconfig/client_gapic.yaml

```diff
@@ -0,0 +1,6 @@
+type: com.google.api.codegen.ConfigProto
+templates:
+- codegen_provider: com.google.api.codegen.clientconfig.ClientConfigGapicProvider
+  input_element_view: com.google.api.codegen.InterfaceView
+  snippet_files:
+  - main.snip
```

Comment:
That's kind of what I was thinking - it didn't look like this scenario needed much specialized bootstrapping or context support in comparison to other cases. Could we rename main.snip to json.snip? (or maybe main_json.snip?)